starlite/handlers/websocket_handlers.py

```diff
@@ -99,3 +112,252 @@ def _validate_handler_function(self) -> None:
+            name: A string identifying the route handler.
+            on_accept: Callback invoked after a connection has been accepted, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            on_disconnect: Callback invoked after a connection has been closed, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            opt: A string keyed mapping of arbitrary values that can be accessed in :class:`Guards <.types.Guard>` or
+                wherever you have access to :class:`Request <.connection.Request>` or
+                :class:`ASGI Scope <.types.Scope>`.
+            signature_namespace: A mapping of names to types for use in forward reference resolution during signature
+                modelling.
+            type_encoders: A mapping of types to callables that transform them into types supported for serialization.
+            **kwargs: Any additional kwarg - will be set in the opt dictionary.
+        """
+        self._receive_mode = receive_mode
+        self._send_mode = send_mode
+        self._on_accept = AsyncCallable(on_accept) if on_accept else None
+        self._on_disconnect = AsyncCallable(on_disconnect) if on_disconnect else None
+        self.type_encoders = type_encoders
+
+        super().__init__(
+            path=path,
+            dependencies=dependencies,
+            exception_handlers=exception_handlers,
+            guards=guards,
+            middleware=middleware,
+            name=name,
+            opt=opt,
+            signature_namespace=signature_namespace,
+            **kwargs,
+        )
+
+    def _validate_handler_function(self) -> None:
+        """Validate the route handler function once it's set by inspecting its return annotations."""
+
+    def _create_listener_fn(  # noqa: C901
+        self,
+        *,
+        wants_receive_type: Any,
+        can_send_data: bool,
+        may_not_send_data: bool,
+        should_encode_to_json: bool,
+        pass_socket: bool,
+        callback: AsyncCallable,
+    ) -> Callable[..., Coroutine[None, None, None]]:
+        json_encoder = JsonEncoder(enc_hook=partial(default_serializer, type_encoders=self.resolve_type_encoders()))
+
+        async def handle_receive(socket: WebSocket) -> Any:
+            received_data = await socket.receive_data(mode=self._receive_mode)  # pyright: ignore
+            if wants_receive_type is str:
+                if isinstance(received_data, bytes):
+                    received_data = received_data.decode("utf-8")
+            elif wants_receive_type is bytes:
+                if isinstance(received_data, str):
+                    received_data = received_data.encode("utf-8")
+            else:
+                received_data = decode_json(received_data)
+            return received_data
+
+        async def handle_send(socket: WebSocket, data_to_send: Any) -> None:
+            if not can_send_data:
```

Comment:
I'm not sure if we should do this runtime validation, as we don't have this in place everywhere else. @starlite-api/maintainers?

Reply:
im not in favor

Reply:
Then let's not